File: app/src/main/java/net/squanchy/tweets/domain/TweetLinkInfo.java
package net.squanchy.tweets.domain;

import com.google.auto.value.AutoValue;
import com.twitter.sdk.android.core.models.Tweet;

@AutoValue
public abstract class TweetLinkInfo {

    public static TweetLinkInfo from(Tweet tweet) {
        return new AutoValue_TweetLinkInfo(tweet.idStr, tweet.user.screenName);
    }

    public abstract String statusId();

    public abstract String screenName();
}

File: app/src/main/java/net/squanchy/tweets/domain/TweetLinkInfo.kt
package net.squanchy.tweets.domain

import net.squanchy.service.firebase.model.twitter.FirestoreTweet

data class TweetLinkInfo(private val tweet: FirestoreTweet) {
    val statusId: String
        get() = tweet.id
    val screenName: String
        get() = tweet.user.screenName
}
